examples/motor-imagery/helpers/frequencybandselection_helpers.py

```diff
@@ -0,0 +1,226 @@
+        Id of the events to consider.
+
+        - If dict, the keys can later be used to access associated
+          events.
+        - If int, a dict will be created with the id as string.
+        - If a list, all events with the IDs specified in the list
+          are used.
+        - If None, all events will be used and a dict is created
+          with string integer names corresponding to the event id integers.
+    cv : cross-validation generator, default=None
+        An instance of a cross validation iterator from sklearn.
+        If method is "cross_validation", cv needs to be defined.
+    train_ind : ndarray, shape (n_training_set,), default=None
+        index of training set.
+        If method is "train_test_split", train_ind needs to be defined.
+    method : str, default="cross_validation",
+        Method of data split either cross_validation or train_test_split.
+    return_class_dis : bool, default=False
+        Whether to return class_dis value.
+    verbose : bool, str, int, default=None
+        Control verbosity of the logging output of filtering and .
+        If None, use the default verbosity level.
+
+    Returns
+    -------
+    best_freqs : list
+        List of the selected frequency band for each hold of
+        cross-validation, if method is "cross_validation"; selected
+        frequency band if method is "train_test_split".
+    class_dists : list
+        List of class_dis value of each hold of cross validation,
+        if method is "cross_validation"; list of class_dis value
+        if method is "train_test_split".
+
+    Notes
+    -----
+    .. versionadded:: 0.3.1
+
+    References
+    ----------
+    .. [1] `Class-distinctiveness-based frequency band selection on the
+       Riemannian manifold for oscillatory activity-based BCIs: preliminary
+       results
+       <https://hal.archives-ouvertes.fr/hal-03641137/>`_
+       M. S. Yamamoto, F. Lotte, F. Yger, and S. Chevallier.
+       44th Annual International Conference of the IEEE Engineering
+       in Medicine & Biology Society (EMBC2022), 2022.
+    """
+
+    subband_fmin = list(np.arange(freq_band[0],
+                                  freq_band[1] - sub_band_width + 1.,
+                                  sub_band_step))
+    subband_fmax = list(np.arange(freq_band[0] + sub_band_width,
+                                  freq_band[1] + 1., sub_band_step))
+    nb_subband = len(subband_fmin)
+
+    all_sub_band_cov = []
+
+    for ii in range(nb_subband):
+        cov_data, labels = _get_filtered_cov(raw, picks,
+                                             event_id,
+                                             subband_fmin[ii],
+                                             subband_fmax[ii],
+                                             tmin, tmax, verbose)
+        all_sub_band_cov.append(cov_data)
+
+    all_cv_best_freq = []
+    all_cv_class_dis = []
+
+    if method == 'cross_validation':
```

Comment:
I think you can avoid this if / else that duplicates a lot of logic. Just use CV as mentioned above

Reply:
I removed the if/else. Now, the function has become more concise.